v1.1/parallel-statement-execution.md

```diff
@@ -0,0 +1,128 @@
+
+As seen earlier, sequential execution of SQL statements returns a return value immediately from the server. The next statement is executed only after the return value is received. Appending the `RETURNING NOTHING` clause with SQL statements prevents the server from sending return values and allows CockroachDB to execute the statements in parallel. To execute statements in parallel, append the `RETURNING NOTHING` suffix to the SQL statements. The statements are executed in parallel till CockroachDB encounters a **barrier statement**. A barrier statement is a statement that needs to be executed sequentially.
+
+In our example, the transaction would be as follows:
+
+~~~ sql
+> BEGIN;
+> UPDATE username SET last_name = 'Smith' WHERE id = 1 RETURNING NOTHING;
+> UPDATE movies SET favorite_movie = 'The Matrix' WHERE id = 1 RETURNING NOTHING;
+> UPDATE songs SET favorite_song = 'All this time' WHERE id = 1 RETURNING NOTHING;
+> COMMIT;
+~~~
+
+In this case, because the statements within the transaction are independent of each other, they can be executed in parallel without affecting the results. The COMMIT statement is executed sequentially, and thus acts as a barrier statement. Barrier statements are executed only after all the parallel statements are executed.
+
+The following conceptual diagram shows how the transaction is executed sequentially and in parallel. The diagram also shows how executing statements in parallel reduces the aggregate latency.
+
+<img src="{{ 'images/Parallel_Statement_Normal_Execution.png' | relative_url }}" alt="CockroachDB Parallel Statement Execution" style="border:1px solid #eee;max-width:100%" />
+
+### Perceived delay in execution of barrier statements 
+
+As stated earlier, barrier statements are executed only after all parallel statements are executed. Thus it may seem as if the barrier statement is taking longer to execute, but it's in fact waiting on the parallel statements. Even then, the total time required for parallel execution of statements followed by the sequential execution of the SQL statement should be less than time required for the sequential execution of all statements. 
+
+Referring to the previous diagram, all UPDATE statements are executed before executing COMMIT. Hence it might seem as if COMMIT is taking longer to execute, but it is, in fact, waiting on the UPDATE statements to be executed.
+
+### Error message mismatch
+
+In case of parallel statement execution, it is possible to see a mismatch in the error message being displayed and the statement being executed. 
+
+Because the `RETURNING NOTHING` clause is used with parallel statements, CockroachDB cannot return the error for the parallel statement. The error message for the parallel statement is then reported on the next barrier statement. Thus you might see a mismatch with the error message to the statement being executed. 
```

Comment:
> Thus you might see a mismatch with the error message to the statement being executed. 

Thus, if any parallel statement encounters an error, you **will** see the error message as the result of the barrier statement.

Reply:
Can we say "Thus, if any parallel statement encounters an error, you will see the error message after the barrier statement"?

Reply:
"after" isn't the right word though. It needs to be super clear that the error message will always be returned **as the result of** the barrier statement.

Reply:
"As the result of" is eerily similar to "as a result of". It could be misread as the error message is the result of the barrier statement, which isn't the case. "After" seems more appropriate in that context.